File: Medium/2026-05-13-1674-Minimum-Moves-to-Make-Array-Complementary/solution.java
class Solution {
    public int minMoves(int[] nums, int limit) {
        int n = nums.length;
        int[] moves = new int[2 * limit + 2];
        
        for (int i = 0; i < n / 2; i++) {
            int a = nums[i];
            int b = nums[n - 1 - i];
            int low = Math.min(a, b) + 1;
            int high = Math.max(a, b) + limit;
            
            moves[low]++;
            moves[high + 1]--;
        }
        
        int minMoves = n;
        int currentMoves = 0;
        
        for (int i = 1; i < moves.length; i++) {
            currentMoves += moves[i];
            minMoves = Math.min(minMoves, currentMoves);
        }
        
        return minMoves;
    }
}

File: Medium/2026-05-13-1674-Minimum-Moves-to-Make-Array-Complementary/solution.js
var minMoves = function(nums, limit) {
    const n = nums.length;
    const moves = new Array(n + 2).fill(0);
    
    for (let i = 0; i < n / 2; i++) {
        const a = nums[i];
        const b = nums[n - 1 - i];
        const low = Math.max(2, a + b);
        const high = Math.min(limit + 1, a + b + 1);
        
        moves[2] += 2; // We need at least 2 moves to make them complementary
        moves[low] -= 1; // We can reduce the moves needed starting from low
        moves[high] += 1; // We can increase the moves needed starting from high
    }
    
    let minMoves = Infinity;
    let currentMoves = 0;
    
    for (let i = 2; i <= limit + 1; i++) {
        currentMoves += moves[i];
        minMoves = Math.min(minMoves, currentMoves);
    }
    
    return minMoves;
};